plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeConnectorSmokeTest.java

```diff
@@ -1751,6 +1763,103 @@ public void testOptimizeUsingForcedPartitioning()
         assertThat(getAllDataFilesFromTableDirectory(tableName)).isEqualTo(union(initialFiles, updatedFiles));
     }
 
+    @Test
+    public void testRegisterTableWithTableLocation()
+    {
+        String tableName = "test_register_table_with_table_location_" + randomNameSuffix();
+
+        assertUpdate("CREATE TABLE " + tableName + " (a int, b varchar, c boolean)");
+        assertUpdate("INSERT INTO " + tableName + " VALUES(1, 'INDIA', true)", 1);
+
+        String tableLocation = getTableLocation(tableName);
+        String showCreateTableOld = (String) computeScalar("SHOW CREATE TABLE " + tableName);
+        // Drop table from metastore and use the same table name to register again
+        dropTableFromMetastore(tableName);
+
+        assertUpdate("CALL system.register_table (CURRENT_SCHEMA, '" + tableName + "', '" + tableLocation + "')");
```

Comment:
Let's add `assertQueryFails` after `dropTableFromMetastore` to ensure the table isn't accessible without `register_table` procedure. At least one test should have this check. There's no need to add to other test methods. 

 I would recommend fixing `BaseDeltaLakeRegisterTableProcedureTest` too.